File: Support/getpyencoding.py
#!/usr/bin/env python

import re

while True:
    try:
        m = re.search('#.*?coding[:=]\s*([-\w.]+)', raw_input())
    except EOFError:
        raise SystemExit
    if m is not None:
        print m.groups(1)[0]
        raise SystemExit

File: Support/getpyencoding.pl
#!/usr/bin/perl

# A glorified one-liner. Yay for language promiscuity!

while (<>) {
    last if $. > 2;
    if (/\s*#.*?coding[:=]\s*([-\w.]+)/) {
        print $1;
        last;
    }
}
